pkg/agent/route/route_linux.go

```diff
@@ -59,27 +59,27 @@ var (
 
 // Client takes care of routing container packets in host network, coordinating ip route, ip rule, iptables and ipset.
 type Client struct {
-	nodeConfig  *config.NodeConfig
-	encapMode   config.TrafficEncapModeType
-	serviceCIDR *net.IPNet
-	ipt         *iptables.Client
+	nodeConfig *config.NodeConfig
+	encapMode  config.TrafficEncapModeType
+	ipt        *iptables.Client
 	// nodeRoutes caches ip routes to remote Pods. It's a map of podCIDR to routes.
 	nodeRoutes sync.Map
 	// nodeNeighbors caches IPv6 Neighbors to remote host gateway
 	nodeNeighbors sync.Map
 }
 
 // NewClient returns a route client.
+// Todo: remove param serviceCIDR after kube-proxy is replaced by Antrea Proxy. This param is not used in this file,
+// leaving it here is to be compatible with the implementation on Windows.
 func NewClient(serviceCIDR *net.IPNet, encapMode config.TrafficEncapModeType) (*Client, error) {
 	ipt, err := iptables.New()
```

Comment:
I feel the code will be simpler if we New and set Client.ipt in initIPTables(), then enableIPv4/v6 can be two parameters of iptables.New(), and we can remove SetSupportedIPProtocols().